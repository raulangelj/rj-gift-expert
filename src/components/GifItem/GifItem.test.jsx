import { render, screen } from '@testing-library/react';
import { GifItem } from './GifItem';

describe('Pruebas <GifItem />', () => {
  const title = 'Titulo del GIF';
  const url = 'https://giphy.com/embed/leB1qRInddfWdAe2Yw';

  test('Debe de hacer match con el snapshot', () => {
    const {container } = render(
      <GifItem
        title={title}
        url={url}
      />
    )

    expect(container).toMatchSnapshot();
  })

  test('Debe de mostrar la imagen con el url y el ALT correcto', () => {
    render(
      <GifItem
        title={title}
        url={url}
      />
    )

    const { src, alt } = screen.getByRole('img');

    expect(src).toBe(url);
    expect(alt).toBe(title);
  })

  test('Debe de mostrar el titulo en el componente', () => {
    render(
      <GifItem
        title={title}
        url={url}
      />
    )

    expect(screen.getByText(title)).toBeTruthy();
  })
})