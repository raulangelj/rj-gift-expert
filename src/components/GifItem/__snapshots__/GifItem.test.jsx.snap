// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <GifItem /> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo del GIF"
      src="https://giphy.com/embed/leB1qRInddfWdAe2Yw"
    />
    <p>
      Titulo del GIF
    </p>
  </div>
</div>
`;
